atc/metric/emitter/influxdb.go

```diff
@@ -46,48 +58,82 @@ func (config *InfluxDBConfig) NewEmitter() (metric.Emitter, error) {
 	}
 
 	return &InfluxDBEmitter{
-		client:   client,
-		database: config.Database,
+		Client:   	client,
+		Database: 	config.Database,
+		BatchSize: 	int(config.BatchSize),
+		BatchDuration: 	config.BatchDuration,
 	}, nil
 }
 
-func (emitter *InfluxDBEmitter) Emit(logger lager.Logger, event metric.Event) {
+func emitBatch(emitter *InfluxDBEmitter, logger lager.Logger, events []metric.Event) {
+
+	logger.Debug("influxdb-emit-batch", lager.Data{
+		"size": len(events),
+	})
 	bp, err := influxclient.NewBatchPoints(influxclient.BatchPointsConfig{
-		Database: emitter.database,
+		Database: emitter.Database,
 	})
 	if err != nil {
 		logger.Error("failed-to-construct-batch-points", err)
 		return
 	}
 
-	tags := map[string]string{
-		"host": event.Host,
+	for _, event := range events {
+		tags := map[string]string{
+			"host": event.Host,
+		}
+
+		for k, v := range event.Attributes {
+			tags[k] = v
+		}
+
+		point, err := influxclient.NewPoint(
+			event.Name,
+			tags,
+			map[string]interface{}{
+				"value": event.Value,
+				"state": string(event.State),
+			},
+			event.Time,
+		)
+		if err != nil {
+			logger.Error("failed-to-construct-point", err)
+			continue
+		}
+
+		bp.AddPoint(point)
 	}
 
-	for k, v := range event.Attributes {
-		tags[k] = v
-	}
-
-	point, err := influxclient.NewPoint(
-		event.Name,
-		tags,
-		map[string]interface{}{
-			"value": event.Value,
-			"state": string(event.State),
-		},
-		event.Time,
-	)
-	if err != nil {
-		logger.Error("failed-to-construct-point", err)
-		return
-	}
-
-	bp.AddPoint(point)
-
-	err = emitter.client.Write(bp)
+	err = emitter.Client.Write(bp)
 	if err != nil {
 		logger.Error("failed-to-send-points",
 			errors.Wrap(metric.ErrFailedToEmit, err.Error()))
 		return
 	}
+	logger.Info("influxdb-emitter-fork-influxdb-batch-emitted", lager.Data{
```

Comment:
just noticed this log line :sweat_smile: looks like more dev logging? (noticed the 'fork')

Reply:
Thanks for catching this @vito. It was definitely dev logging. Removed and just waiting for the tests to pass.